src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -384,18 +385,37 @@
 	 * @return a {@link HttpClient}
 	 */
 	public static HttpClient create() {
-		return create(HttpResources.get());
+		return new HttpClientConnect(new HttpConnectionProvider(HttpResources.get(), Http2Resources::get));
 	}
 
 	/**
 	 * Prepare an {@link HttpClient}. {@link UriConfiguration#uri(String)} or
 	 * {@link #baseUrl(String)} should be invoked before a verb
 	 * {@link #request(HttpMethod)} is selected.
 	 *
+	 * @param connectionProvider the {@link ConnectionProvider} to be used
 	 * @return a {@link HttpClient}
 	 */
 	public static HttpClient create(ConnectionProvider connectionProvider) {
-		return new HttpClientConnect(connectionProvider);
+		Objects.requireNonNull(connectionProvider, "connectionProvider");
+		return new HttpClientConnect(new HttpConnectionProvider(connectionProvider));
+	}
+
+	/**
+	 * Prepare an {@link HttpClient}. {@link UriConfiguration#uri(String)} or
+	 * {@link #baseUrl(String)} should be invoked before a verb
+	 * {@link #request(HttpMethod)} is selected.
+	 *
+	 * @param connectionProvider the {@link ConnectionProvider} to be used
+	 * @param maxHttp2Connections the max number of connections that will be used for HTTP/2 requests
+	 * @return a {@link HttpClient}
+	 */
+	public static HttpClient create(ConnectionProvider connectionProvider, int maxHttp2Connections) {
```

Comment:
This might be a bit limiting - is there any other configuration that might apply later ? E.g. connection timeout etc. Should this be a Supplier of PoolProvider or similar ? or a Consumer of a spec?

Reply:
we will think about this in M2